app/src/main/java/nerd/tuxmobil/fahrplan/congress/schedule/FahrplanFragment.java

```diff
@@ -249,38 +257,37 @@ public void onResume() {
         fillTimes();
     }
 
-    private void viewDay(boolean reload) {
-        Log.d(LOG_TAG, "viewDay(" + reload + ")");
+    private void viewDay(boolean forceReload) {
+        Log.d(LOG_TAG, "viewDay(" + forceReload + ")");
+        View layoutRoot = getView();
+        int boxHeight = getNormalizedBoxHeight(getResources(), scale, LOG_TAG);
 
-        loadLectureList(appRepository, mDay, reload);
-        List<Lecture> lectures = MyApp.lectureList;
-        if (lectures != null && !lectures.isEmpty()) {
-            conference.calculateTimeFrame(lectures, dateUTC -> new Moment(dateUTC).getMinuteOfDay());
-            MyApp.LogDebug(LOG_TAG, "Conference = " + conference);
+        HorizontalSnapScrollView horizontalScroller = layoutRoot.findViewById(R.id.horizScroller);
+        if (horizontalScroller != null) {
+            horizontalScroller.scrollTo(0, 0);
         }
-        View layoutRoot = getView();
-        HorizontalSnapScrollView scroller = layoutRoot.findViewById(R.id.horizScroller);
-        if (scroller != null) {
-            scroller.scrollTo(0, 0);
-            addRoomColumns(scroller);
+
+        loadLectureList(appRepository, mDay, forceReload);
+        List<Lecture> lecturesOfDay = MyApp.lectureList;
+
+        if (lecturesOfDay != null) {
+            if (!lecturesOfDay.isEmpty()) {
+                conference.calculateTimeFrame(lecturesOfDay, dateUTC -> new Moment(dateUTC).getMinuteOfDay());
+                MyApp.LogDebug(LOG_TAG, "Conference = " + conference);
+            }
+            if (horizontalScroller != null) {
+                if (horizontalScroller.getColumnWidth() != 0) {
+                    // update pre-calculated roomColumnWidth with actual layout
+                    roomColumnWidth = horizontalScroller.getColumnWidth();
```

Comment:
:mag: Is the precalculation in `onViewCreated` obsolete? It looks as if the value is updated here anyway.

Reply:
I thought the same, but was not confident enough to remove it now. Maybe in another "FahrplanFragment cleanup" pull request.  It was there before already, so I will leave it there for now.

Reply:
Okay. But please add a `// TODO Assignment might be obsolete. Remove if verified.` comment there so one can find the cleanup task.